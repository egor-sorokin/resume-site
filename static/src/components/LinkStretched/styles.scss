@import '../../styles/base/_vars.scss';

.link-stretched {
  letter-spacing: 3px;

  transform: translateX(-40px);
  transition: transform $defaultTiming $defaultEasing,
  letter-spacing $defaultTiming $defaultEasing;
}

.link-stretched:hover {
  letter-spacing: 5px;

  transform: translateX(-34px);
}
