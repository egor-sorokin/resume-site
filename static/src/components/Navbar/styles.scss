@import '../../styles/base/_vars.scss';

.navbar {
  width: 100%;

  &__item {
    position: relative;

    &:nth-of-type(n+2) {
      margin-top: 6px;
    }
    &:nth-of-type(2n) {
      left: 55%;
      text-align: left;
    }
    &:nth-of-type(2n+1) {
      left: -55%;
      text-align: right;
    }
  }
  &__text {
    color: $mercuryLight;
    font-family: $secondaryFont;
    font-size: 12px;
    letter-spacing: 3.4px;
    text-transform: uppercase;
    font-weight: 500;
  }
}
