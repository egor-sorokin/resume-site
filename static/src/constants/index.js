export const DEFAULT_API_URL = 'http://127.0.0.1:8080/api/v.1.0/';
export const URL_PATH_PERSON_DATA = 'person';
export const URL_PATH_PROJECTS = 'projects';
