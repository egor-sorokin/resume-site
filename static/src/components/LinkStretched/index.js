import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';


const LinkStretched = props => {
  const {text, url, target, cssClasses, onClick} = props;

  const handleClick = (e) => {
    if (!url) {
      e.preventDefault();
      onClick();
    } else {
      onClick(e.currentTarget.dataset.anchorId.substring(1), true);
    }
  };

  return (
    <a
      href={url}
      data-anchor-id={url}
      target={target}
      onClick={handleClick}
      className={'link-stretched ' + cssClasses}
    >{text}</a>
  )
};


LinkStretched.propTypes = {
  onClick: PropTypes.func,
  cssClasses: PropTypes.string,
  text: PropTypes.string,
  url: PropTypes.string,
  target: PropTypes.string
};

LinkStretched.defaultProps = {
  cssClasses: '',
  text: '',
  url: '',
  target: '',
  onClick: () => {
  }
};


export default LinkStretched;