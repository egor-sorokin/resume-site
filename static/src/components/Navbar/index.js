import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';


const Navbar = ({items}) => {
  const navbarItems = items.map((item) => (
    <li className="navbar__item">
      <a href="/" className="navbar__text">{item}</a>
    </li>
  ));

  return (
    <ul className="navbar">
      {navbarItems}
    </ul>
  )
};

Navbar.propTypes = {
  items: PropTypes.instanceOf(Array)
};

Navbar.defaultProps = {
  items: []
};

export default Navbar;