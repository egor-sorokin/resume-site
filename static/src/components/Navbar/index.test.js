// eslint-disable-next-line
import {shallow, render, mount} from 'enzyme';
import React from 'react';
import LinkStretched from '../LinkStretched/index';
import Navbar from './index';
import {NAVBAR_ITEMS} from '../../constants/index';


describe('<Navbar />', () => {
  let component;
  let scrollToSection;
  const section = 'home';


  describe('base tests', () => {
    beforeEach(() => {
      component = shallow(<Navbar />);
    });


    it('renders without crashing', () => {
      expect(component).toBeDefined();
    });


    it('matches snapshot', () => {
      expect(component).toMatchSnapshot();
    });


    it('renders 0 component', () => {
      expect(component.children()).toHaveLength(0);
    });
  });


  describe('link functionality tests', () => {
    beforeEach(() => {
      scrollToSection = jest.fn(() => 'mock implementation');
    });


    it('clicks on an anchor link', () => {
      component = mount(<Navbar items={NAVBAR_ITEMS}/>);

      const linkButton = component.find(LinkStretched).first().find('a').first();
      linkButton.simulate('click', section);

      expect(scrollToSection).toHaveBeenCalled();
      expect(scrollToSection).toHaveBeenCalledTimes(1);
    });
  });
});
