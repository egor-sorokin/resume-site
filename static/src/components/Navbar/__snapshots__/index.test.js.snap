// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> base tests matches snapshot 1`] = `
<ul
  className="navbar"
/>
`;
